### DISCUSSION TITLE
Replacing facet with map/reduce

### DISCUSSION THREAD

**Discussion by instantsquid** [ravendb/ravendb#16754] [General]
Firstly an apology - our Production system is still running on V3.5 - not my decision, it is what it is.
I have inherited a system that implements a faceted search feature, with around 1.5 million documents in the database. There are around a dozen facets, some of which have just a handful of values, others have a couple of hundred, and one has several thousand. A bug has been reported which I've been tasked with fixing, that is caused by the MaxPageSize server setting (default 1024). Essentially, one facet has several thousand distinct values so the results are truncated.
Apart from the sledge-hammer option of increasing MaxPageSize, I've found limited resources online as to how else this could be addressed. One StackOverflow post from 2011 hints that the correct approach would be to use a map/reduce index. I have read the documentation, but I cannot see how a facet could be implemented using an index.
Happy to provide more information if necessary. Thank you in advance.

**Comment by ayende** [comment]
I would recommend raising the limit in the configuration.
If you are on 3.5, I assume that this is a legacy system and won't have enough time to fix things in a proper fashion.

FWIW, RavenDB 5.4 doesn't have the hard limit of 1024 in facets.

**Reply by instantsquid** [reply]
Thanks very much for the swift reply. I'm going to recommend we increase the MaxPageSize in a test environment and pay attention to the performance of the server while we run our automated tests.
